vendor/wp-now/src/wp-now.ts

```diff
@@ -176,6 +178,18 @@ async function prepareWordPress( php: PHP, options: WPNowOptions ) {
 			await runWpPlaygroundMode(php, options);
 			break;
 	}
+
+	// The symlink manager ensures that we mount the targets of symlinks so that they
+	// work inside the php runtime. It also watches for changes to ensure symlinks
+	// are managed correctly.
+	// When the runtime exits, we need to stop the watcher.
+	const symlinkManager = new SymlinkManager(php, options.projectPath);
+	await symlinkManager.scanAndCreateSymlinks();
+	symlinkManager.startWatching();
+	php.addEventListener('runtime.beforedestroy', () => {
+		console.log('Stopping symlink manager');
+		symlinkManager.stopWatching();
+	})
```

Comment:
This may be another issue. I assumed that when we stopped the site (stop the server) it would also clean up the PHP instance (exiting it) but it does not. We should always clean up resources when they are no longer needed as not doing so may lead to memory leaks.

I think the locality of the symlink manager makes it unwieldy to hook into the stop server/stop site callback. Symlinks are closely related to the PHP instance and are recreated on rotation of the instance. I think it is better to hook into the lifecycle events of the PHP instance.

The other option to consider is to reuse the same Symlink manager instance when PHP rotates and have a way to remount the symlink targets on rotation by using the same instance of the symlink manager. We could then return the instance of the symlink manager and use in the `stopServer` method. I prefer an event-based approach though.

What do you think? Should we add a `php.exit()` call when we stop the server?

Reply:
Oh, good catch! We probably overlooked destroying the PHP instance in this logic, which as you pointed out, might be producing memory leaks.

> What do you think? Should we add a `php.exit()` call when we stop the server?

Yes, that makes total sense.  In some cases, I experienced that if the site is busy processing PHP requests, the site takes seconds to stop. This will probably improve that issue.